docs/permission/writing-a-policy.md

```diff
@@ -84,59 +98,68 @@ In addition to the conditions provided by the catalog plugin, you can write your
 +         catalogConditions.isEntityOwner(
 +           user?.identity.ownershipEntityRefs ?? [],
 +         ),
 +       );
 +     }
-
-      return { result: AuthorizeResult.ALLOW };
     }
+    return { result: AuthorizeResult.ALLOW };
+  }
 ```
 
-If you choose to write your policy this way, it will certainly work! You should be able to verify this by saving this policy and seeing that the catalog now only shows the entities that you own. However, you can imagine that as policies grow to handle many different permissions, these conditionals can quickly become repetitive. We can author this same policy in a more scalable way by using resource types.
+If you choose to write your policy this way, it will certainly work! You should be able to verify this by saving this policy and seeing that the catalog now only shows the entities that you own. However, you can imagine that as policies grow to handle many different permissions, these conditionals can quickly become repetitive. We can author this same policy in a more scalable way by using permission attributes.
 
 ```diff
-    async handle(
-      request: PolicyAuthorizeQuery,
-      user?: BackstageIdentityResponse,
-     ): Promise<PolicyDecision> {
--     if (request.permission.name === 'catalog.entity.delete') {
-+     if (request.permission.resourceType === 'catalog-entity') {
-        return createCatalogPolicyDecision(
+  async handle(
+    request: PolicyQuery,
+    user?: BackstageIdentityResponse,
+  ): Promise<PolicyDecision> {
+    if (isResourcePermission(request.permission, 'catalog-entity')) {
+-     if (request.permission.name === 'catalog.entity-delete') {
++     if (
++       request.permission.attributes.action === 'delete' ||
++       request.permission.attributes.action === 'read'
++     ) {
+        return createCatalogConditionalDecision(
+          request.permission,
           catalogConditions.isEntityOwner(
             user?.identity.ownershipEntityRefs ?? [],
           ),
         );
       }
 
 -     if (request.permission.name === 'catalog.entity.read') {
--       return createCatalogPolicyDecision(
+-       return createCatalogConditionalDecision(
+-         request.permission,
 -         catalogConditions.isEntityOwner(
 -           user?.identity.ownershipEntityRefs ?? [],
 -         ),
 -       );
 -     }
--
-      return { result: AuthorizeResult.ALLOW };
     }
+    return { result: AuthorizeResult.ALLOW };
+  }
 ```
 
-In this example, we use the `catalog-entity` resource type to catch all authorization requests that have to do with resources from the catalog. Now, you should be able to see the same functionality as before (only see the catalog entities that you own) - success!
+In this example, we use the `action` permission attribute to catch all authorization requests that have to do with `read` and `delete` permissions. Now, you should be able to see the same functionality as before (only see the catalog entities that you own) - success!
```

Comment:
I think this fundamentally changes the policy that we're crafting. Is there a reason we switched to `read` and `delete` permissions instead of `catalog-entity` resource type?

Reply:
Resolving since this is resolved by latest commit